website/content/docs/configuration/listener/tcp.mdx

```diff
@@ -213,6 +213,62 @@ default value in the `"/sys/config/ui"` [API endpoint](/api-docs/system/config-u
   there is no X-Forwarded-For header or it is empty, the client address will be
   used as-is, rather than the client connection rejected.
 
+### ACME parameters
+
+The following parameters replace direct certificate acquisition when
+`tls_cert_file` is empty. ACME defaults to being enabled. Set the
+`tls_acme_ca_directory` to the empty string to disable. These options belong
+in the root of the TCP listener configuration.
+
+- `tls_acme_ca_directory` `(string: "https://acme-v02.api.letsencrypt.org/directory")` - Path
+  to the ACME directory; defaults to Let's Encrypt's production CA.
+
+- `tls_acme_cache_path` `(string: "~/.local/share/certmagic")` - Specifies the
```

Comment:
This default implies that the (linux) user running OpenBao has a home directory. In our Docker images and DEB/RPM packages, openbao is running as a system user without a home directory. So accessing the default path might return an error. 

Not sure what a sensible default might be though. Maybe `/var/lib/openbao/certmagic` :thinking:

Reply:
> This default implies that the (linux) user running OpenBao has a home directory. In our Docker images and DEB/RPM packages, openbao is running as a system user without a home directory. So accessing the default path might return an error.

Hmmm, I'm not sure how we'd be able to conditionally check if we can use any other path, tbh. This is also not our default, but certmagic's default: https://pkg.go.dev/github.com/caddyserver/certmagic#readme-storage

`/var/lib/openbao` may not exist, and so the `openbao` user may not have permissions to write to it. I think that means we'd have to fall back to `$TEMPDIR`?

Reply:
Interesting, the certmagic default is subtly different from what you wrote

> By default, CertMagic stores assets on the local file system in $HOME/.local/share/certmagic (and honors $XDG_DATA_HOME if set). CertMagic will create the directory if it does not exist. If writes are denied, things will not be happy, so make sure CertMagic can write to it!

I guess this is fine then